libsolidity/interface/CompilerStack.cpp

```diff
@@ -1045,6 +1045,16 @@ void CompilerStack::compileContract(
 		solAssert(false, "Assembly exception for deployed bytecode");
 	}
 
+	// Throw an error if EIP-170 limits are exceeded.
+	if (
+		m_evmVersion >= langutil::EVMVersion::spuriousDragon() &&
+		compiledContract.object.bytecode.size() > 0x6000
```

Comment:
Only checking object here because that includes runtimeObject.

Reply:
Why not check runtimeObject?

Reply:
Because `object` includes `runtimeObject`. Is there a case where `runtimeObject` exceeds the size, but `object` doesn't? Also note in the code above both are always compiled.

Reply:
I can check both though.

Reply:
Is there a restriction on the tx payload? I thought only on the deployed size.

Reply:
Hm, good question, I thought it should apply to the tx payload, because the whole point is to be able to throw away oversized transactions from the mempool.

Reply:
> If block.number >= FORK_BLKNUM, then if contract creation initialization returns data with length of more than 0x6000 (2**14 + 2**13) bytes, contract creation fails with an out of gas error.

You are correct. The `runtimeObject` has to be checked, the init code can be arbitrarily large.

Reply:
Fixed.